challenge_one/src/agent.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+} from "forta-agent";
+
+export const NETHERMIND_FORTA_ADDRESS: string =
+  "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8";
+
+export const UPDATE_AGENT_FUNCTION: string =
+  "function updateAgent(uint256 agentId,string metadata,uint256[] chainIds)";
+export const CREATE_AGENT_FUNCTION: string =
+  "function createAgent(uint256 agentId,address owner,string metadata,uint256[] chainIds)";
+
+export const provideHandleTransaction =
+  (): HandleTransaction =>
+  async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    if (NETHERMIND_FORTA_ADDRESS === txEvent.from) {
+      return findings;
+    }
+
+    // filter the transaction logs for Bot Update Function call
+    const nethermindBotUpdatesCalls = txEvent.filterFunction(
+      UPDATE_AGENT_FUNCTION
+    );
+
+    // filter the transaction logs for Bot Create Function call
+    const nethermindCreateBotCalls = txEvent.filterFunction(
+      CREATE_AGENT_FUNCTION
+    );
```

Comment:
`filterFunction()` can take a 2nd argument, which is the address of the contract called. Right now, the bot checks all `createAgent`/`updateAgent` function calls, while we need to monitor only those to Forta.